src/bson_mapper/test/bson_streambuf.cpp

```diff
@@ -50,7 +64,7 @@ TEST_CASE("bson_streambuf can faithfully transfer a document",
     int len = bson_view.length();
 
     // set up callback
-    doc_validator_callback cb(bson_obj);
+    auto cb = document_callback(doc_validator_callback(bson_obj));
```

Comment:
If you need the `document_callback` here, I think it would be better to just pass in a lambda, e.g.:
```c++
doc_validator_callback validator(bson_obj);
bson_output_streambuf b_buff([&validator](bsoncxx::document::value v) {
    validator.check(v);
});
```